src/rubyfmt.rb

```diff
@@ -716,6 +716,12 @@ def format_int(ps, rest)
   ps.emit_indent if ps.start_of_line.last
 
   int = rest[0]
+  int_chars = int.chars
+
+  if int_chars[0] != "0" && !int_chars.detect { |c| c == "_" }
+    int = int.chars.reverse.each_slice(3).map(&:join).join("_").reverse
```

Comment:
We could reuse the `int_chars` variable here, instead of calling `int.chars` again